src/main.rs

```diff
@@ -214,6 +227,174 @@ async fn versions(State(_state): State<Arc<AppState>>) -> Result<Json<Value>, St
+        tracing::error!("Query params not provided.");
+        return (
+            StatusCode::BAD_REQUEST,
+            [("Cache-Control", "max-age=60, must-revalidate".to_string())],
+            Json(json!({ "error": "Invalid request: Query params not provided"})),
+        );
+    }
+
+    let cloned_state = state.clone();
+    let mut conn = cloned_state
+        .connection_pool
+        .get_timeout(Duration::from_secs(1))
+        .expect("Get connection pool");
+
+    // Initialize the result variables
+    let mut transaction_results: TransactionResult = TransactionResult::default();
+
+    // Return the combined results
+    if let Some(eth_address) = address_query.eth_address {
+        let ethereum_sends_results = ethereum_sends
+            .select(EthereumSend::as_select())
+            .filter(schema::ethereum_sends::depositor_address.eq(format!("{:?}", eth_address)))
+            .order_by(schema::ethereum_sends::source_timestamp.desc())
+            .limit(500)
+            .load::<EthereumSend>(&mut conn);
+
+        match ethereum_sends_results {
+            Ok(transaction) => {
+                transaction_results.eth_send = transaction
+                    .into_iter()
+                    .map(map_ethereum_send_to_transaction_result)
+                    .collect()
+            }
+            Err(e) => {
+                tracing::error!("Cannot get ethereum send transactions: {:?}", e);
+                return (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    [("Cache-Control", "max-age=60, must-revalidate".to_string())],
+                    Json(json!({ "error": e.to_string()})),
+                );
+            }
+        }
+    }
+    if let Some(avail_address) = address_query.avail_address {
+        let avail_sends_results = avail_sends
+            .select(AvailSend::as_select())
+            .filter(schema::avail_sends::depositor_address.eq(format!("{:?}", avail_address)))
+            .order_by(schema::avail_sends::source_timestamp.desc())
+            .limit(500)
+            .load::<AvailSend>(&mut conn);
+
+        match avail_sends_results {
+            Ok(transaction) => {
+                transaction_results.avail_send = transaction
+                    .into_iter()
+                    .map(map_avail_send_to_transaction_result)
+                    .collect()
+            }
+            Err(e) => {
+                tracing::error!("Cannot get avail send transactions: {:?}", e);
+                return (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    [("Cache-Control", "max-age=60, must-revalidate".to_string())],
+                    Json(json!({ "error": e.to_string()})),
+                );
+            }
+        }
+    }
+
+    (
+        StatusCode::OK,
+        [(
+            "Cache-Control",
+            format!(
+                "public, max-age={}, immutable",
```

Comment:
`immutable` and `maxage` should be set only when both dest and source vars are set. Otherwise, should be considered fresh for 5 minutes (or some duration less than bridging duration).